tests/testParametersClass.m

```diff
@@ -0,0 +1,233 @@
+            testCase.verifyEqual(params.paramsTable{end, 2:4}, [0, 6.7, 10], 'addParam method not working');
+            testCase.verifySize(params.paramsTable, [5, 8], 'Parameters has wrong dimension'); 
+            testCase.verifyError(@() params.addParam({'Param 6', 0, 1, 2, 'false'}), ?MException) % bad fit type
+            params.addParam({'Param 6', 0, 1, 2, true});
+            testCase.verifyEqual(params.paramsTable{end, 1}, "Param 6", 'addParam method not working');
+            testCase.verifyEqual(params.paramsTable{end, 2:4}, [0, 1, 2], 'addParam method not working');
+            testCase.verifyTrue(params.paramsTable{end, 5}, 'addParam method not working');
+            testCase.verifySize(params.paramsTable, [6, 8], 'Parameters has wrong dimension');          
+            testCase.verifyError(@() params.addParam({'c', 1, 1, 1, true, 'u'}), ?MException);  % 6 values not accepted
+            testCase.verifyError(@() params.addParam({'c', 1, 1, 1, true, 'u', 1}), ?MException);  % 7 values not accepted
+            testCase.verifyError(@() params.addParam({'Param 7', 0, 1, 2, false, 'jeff', -1, 1}), ?MException); % bad prior type
+            testCase.verifyError(@() params.addParam({'Param 7', 0, 1, 2, false, 'uniform', '-1', 1}), ?MException); % bad prior value
+            params.addParam({'Param 7', 0, 1, 2, false, 'jeffreys', -1, 1});
+            testCase.verifyEqual(params.paramsTable{end, 1}, "Param 7", 'addParam method not working');
+            testCase.verifyEqual(params.paramsTable{end, 2:4}, [0, 1, 2], 'addParam method not working');
+            testCase.verifyFalse(params.paramsTable{end, 5}, 'addParam method not working');
+            testCase.verifyEqual(params.paramsTable{end, 6}, "jeffreys", 'addParam method not working');
+            testCase.verifyEqual(params.paramsTable{end, 7:8}, [-1, 1], 'addParam method not working');
+            testCase.verifySize(params.paramsTable, [7, 8], 'Parameters has wrong dimension');
+            params.addParam({'Param 8', 0, 1, 2, false, 'gaussian', -1, 1});
+            testCase.verifyEqual(params.paramsTable{end, 6}, "gaussian", 'addParam method not working');
+        end
+
+        function testRemoveParams(testCase) 
+            params = parametersClass(testCase.parameters(1, :));
+            params.paramsTable = [params.paramsTable; vertcat(testCase.parameters(2:end, :))];
+            paramNames = convertCharsToStrings(testCase.parameters(:, 1));
+            % Checks that parameter can be removed
+            testCase.verifyError(@() params.removeParam(), ?MException);
+            testCase.verifyError(@() params.removeParam('bad name'), ?MException);
+            params.removeParam(1);
+            testCase.verifySize(params.paramsTable, [8, 8], 'Parameters has wrong dimension');
+            testCase.verifyEqual(params.paramsTable{:, 1}, paramNames(2:end, 1), 'removeParam method not working');
+            params.removeParam(8, 7, 6);  % is this intended?
```

Comment:
Might be worth testing if a sequence in ascending order works - does it crash due to indices going beyond the maximum?

Reply:
Nice catch, I will fix